app/assets/javascripts/training/components/training_slide_handler.jsx

```diff
@@ -3,97 +3,94 @@ import PropTypes from 'prop-types';
 import createReactClass from 'create-react-class';
 import { browserHistory } from 'react-router';
 import _ from 'lodash';
-
-import TrainingStore from '../stores/training_store.js';
-import TrainingActions from '../actions/training_actions.js';
-import ServerActions from '../../actions/server_actions.js';
+import { connect } from 'react-redux';
+import { fetchTrainingModule, setSlideCompleted, setCurrentSlide, toggleMenuOpen } from '../../actions/training_actions.js';
 import SlideLink from './slide_link.jsx';
 import SlideMenu from './slide_menu.jsx';
 import Quiz from './quiz.jsx';
+import Notifications from '../../components/common/notifications.jsx';
 
 const md = require('../../utils/markdown_it.js').default({ openLinksExternally: true });
 
-const getState = () => TrainingStore.getState();
-
 const TrainingSlideHandler = createReactClass({
   displayName: 'TrainingSlideHandler',
 
   propTypes: {
     params: PropTypes.object
   },
 
-  mixins: [TrainingStore.mixin],
-
   getInitialState() {
-    return {
-      slide: {},
-      menuIsOpen: false,
-      currentSlide: {},
-      nextSlide: {},
-      previousSlide: {},
-      slides: [],
-      loading: true,
-      enabledSlides: []
-    };
+    return { isCompleting: false };
   },
 
   componentWillMount() {
     const slideId = __guard__(this.props.params, x => x.slide_id);
-    ServerActions.fetchTrainingModule({ module_id: this.moduleId(), current_slide_id: slideId });
-    return this.setSlideCompleted(slideId);
+    this.props.fetchTrainingModule({ module_id: this.moduleId(), current_slide_id: slideId });
   },
 
   componentDidMount() {
     window.addEventListener('keyup', this.handleKeyPress);
   },
 
-  componentWillReceiveProps(newProps) {
-    const { slide_id } = newProps.params;
-    TrainingActions.setCurrentSlide(slide_id);
-    this.setSlideCompleted(slide_id);
-    return this.setState(getState(newProps));
-  },
-
   componentWillUnmount() {
     return window.removeEventListener('keyup', this.handleKeyPress);
   },
 
   setSlideCompleted(slideId) {
     const userId = __guard__(document.getElementById('main'), x => x.getAttribute('data-user-id'));
     if (!userId) { return; }
-    return ServerActions.setSlideCompleted({
+    return this.props.setSlideCompleted({
       slide_id: slideId,
       module_id: this.moduleId(),
       user_id: userId
     });
   },
 
-  storeDidChange() {
-    return this.setState(getState());
+  next() {
+    if (this.props.training.valid) {
+      this.setSlideCompleted(this.props.training.currentSlide.slug);
+      this.props.setCurrentSlide(this.props.training.nextSlide.slug);
+    }
+  },
+
+  prev() {
+    if (this.props.training.valid) {
+      this.setSlideCompleted(this.props.training.currentSlide.slug);
```

Comment:
No need to do this for prev, as it will never be the latest slide visited so it will be a noop on the server.